cloudferrylib/os/actions/cinder_database_manipulation.py

```diff
@@ -62,3 +67,56 @@ def run(self, *args, **kwargs):
                 volume['instance_uuid'] = None
                 volume['attach_status'] = 'detached'
         self.get_resource().deploy(jsondate.dumps(data))
+
+
+class CopyVolumesData(action.Action):
+
+    def run(self, **kwargs):
+        data_from_namespace = kwargs.get(NAMESPACE_CINDER_CONST)
+        if not data_from_namespace:
+            raise RuntimeError(
+                "Cannot read attribute {attribute} from namespace".format(
+                    attribute=NAMESPACE_CINDER_CONST))
+        data = jsondate.loads(data_from_namespace)
+        src_cinder_resource = self.src_cloud.resources.get(
+            utils.STORAGE_RESOURCE)
+        dst_cinder_resource = self.dst_cloud.resources.get(
+            utils.STORAGE_RESOURCE)
+        src_host = src_cinder_resource.get_endpoint_host()
+        dst_host = dst_cinder_resource.get_endpoint_host()
+        self.ssh_attempts = self.cfg.migrate.ssh_connection_attempts
+
+        src_dir = self.get_cinder_folder(src_host, self.src_cloud.getIpSsh(),
+                                         self.cfg.src)
+        dst_dir = self.get_cinder_folder(dst_host, self.dst_cloud.getIpSsh(),
+                                         self.cfg.dst)
+
+        dst_ssh_user = self.cfg.dst.ssh_user
+        runner = remote_runner.RemoteRunner(src_host, self.cfg.src.ssh_user,
+                                            password=self.cfg.src.
+                                            ssh_sudo_password)
+        gateway = self.src_cloud.getIpSsh()
+
+        for volume in data['volumes']:
+            done = False
+            attempts = 0
+            while not done:
+                try:
+                    attempts += 1
+                    with settings(gateway=gateway,
+                                  connection_attempts=self.ssh_attempts):
+                        runner.run('rsync --inplace -a -e "ssh -o '
+                                   'StrictHostKeyChecking=no" %s/volume-%s '
+                                   '%s@%s:%s' %
+                                   (src_dir, volume['id'], dst_ssh_user,
+                                    dst_host, dst_dir))
+                    done = True
+                except Exception as e:
+                    if attempts >= COPYING_ATTEMPTS:
```

Comment:
Number of attempts should be configurable. Also, please consider reusing existing retry code or moving this functionality into a separate entity and cover it with unit tests.